src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AwsLambda/HandlerMethodWrapper.cs

```diff
@@ -36,46 +36,102 @@ public AfterWrappedMethodDelegate BeforeWrappedMethod(InstrumentedMethodCall ins
+                    {
+                        agent.Logger.Log(Agent.Extensions.Logging.Level.Warn, $"Found multiple Lambda contexts, will use the first found");
+                    }
+                }
+                else
+                {
+                    var fullName = arg.GetType().FullName;
+                    agent.Logger.Log(Agent.Extensions.Logging.Level.Debug, $"input object fullname = {fullName}");
+                    eventType = fullName.ToEventType();
+                    if (eventType != AwsLambdaEventType.Unknown)
+                    {
+                        inputObject = arg;
+                        agent.Logger.Log(Agent.Extensions.Logging.Level.Debug, $"eventType = {eventType}");
+                    }
+                }
+            }
+
+            string requestId = null;
+            string arn = null;
+            string functionName = null;
+            string functionVersion = null;
+
+            if (lambdaContext != null)
+            {
+                try
+                {
+                    var functionNameGetter = _getFunctionNameFromLambdaContext ??= VisibilityBypasser.Instance.GeneratePropertyAccessor<string>(lambdaContext.GetType(), "FunctionName");
+                    var functionVersionGetter = _getFunctionVersionFromLambdaContext ??= VisibilityBypasser.Instance.GeneratePropertyAccessor<string>(lambdaContext.GetType(), "FunctionVersion");
+                    var requestIdGetter = _getAwsRequestIdFromLambdaContext ??= VisibilityBypasser.Instance.GeneratePropertyAccessor<string>(lambdaContext.GetType(), "AwsRequestId");
+                    var functionArnGetter = _getInvokedFunctionArnFromLambdaContext ??= VisibilityBypasser.Instance.GeneratePropertyAccessor<string>(lambdaContext.GetType(), "InvokedFunctionArn");
 
-            var fullName = inputObject.GetType().FullName;
-            agent.Logger.Log(Agent.Extensions.Logging.Level.Debug, $"input object fullname = {fullName}");
-            var eventType = fullName.ToEventType();
-            agent.Logger.Log(Agent.Extensions.Logging.Level.Debug, $"eventType = {eventType}");
+                    requestId = requestIdGetter(lambdaContext);
 
-            var lambdaFunctionName = functionNameGetter(lambdaContext);
-            var lambdaFunctionArn = functionArnGetter(lambdaContext);
-            var lambdaFunctionVersion = functionVersionGetter(lambdaContext);
+                    functionName = functionNameGetter(lambdaContext);
+                    arn = functionArnGetter(lambdaContext);
+                    functionVersion = functionVersionGetter(lambdaContext);
+                }
+                catch (Exception ex)
+                {
+                    agent.Logger.Log(Agent.Extensions.Logging.Level.Error, $"Failed to read lambda context: {ex.Message}");
+                }
+            }
+            else
+            {
+                agent.Logger.Log(Agent.Extensions.Logging.Level.Warn, $"No Lambda context information found");
```

Comment:
We might want to ensure that this is only logged once, otherwise every invocation can trigger this log message.

Reply:
Sure, we can do that